src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicConfiguration.cs

```diff
@@ -199,6 +199,15 @@ private static MsQuicConfigurationSafeHandle Create(QuicConnectionOptions option
 
     private static unsafe MsQuicConfigurationSafeHandle CreateInternal(QUIC_SETTINGS settings, QUIC_CREDENTIAL_FLAGS flags, X509Certificate? certificate, ReadOnlyCollection<X509Certificate2>? intermediates, List<SslApplicationProtocol> alpnProtocols, QUIC_ALLOWED_CIPHER_SUITE_FLAGS allowedCipherSuites)
     {
+        if (!MsQuicApi.UsesSChannelBackend && certificate is X509Certificate2 cert && intermediates is null)
+        {
+            // MsQuic will not lookup intermediates in local CA store if not explicitly provided,
+            // so we build the cert context to get on feature parity with SslStream. Note that this code
+            // path runs after the MsQuicConfigurationCache check.
```

Comment:
Why is it important to run this after the cache check? I'm not saying it's wrong, I just don't see why this order is the correct one.

Reply:
it avoids perf hit of repeated SslStreamCertificateContext.Create. The cost will be paid only the first time when creating the MsQuicConfiguration.